src/sentry/auth/helper.py

```diff
@@ -457,6 +467,21 @@ def handle_unknown_identity(
             op = None
 
         if not op:
+            if self.user.is_authenticated:
+                template = "auth-confirm-link"
+                existing_user = self.user
+            else:
+                existing_user = acting_user or self._get_user(identity)
```

Comment:
the acting_user will always be None in the case where we'd like to have this new case run. could we simplify by removing this `existing_user` declaration line, and checking if `acting_user` is `None` instead?

Reply:
I think there are two other strategies to try and refactor this area: 

1. use a return early pattern
2. in the above where we set the acting_user to None if there's no password, we could instead create a new "op" for the confirm-account scenario (or re-use confirm and check if the user is authed). that would have the advantage of not having to redefine `user`, and removes a bit of complexity here. There might be a bit of duplication from defining the form logic again, but maybe that could be cleaned up too.

Reply:
Yeah, I wasn't convinced that `acting_user or` did anything but it seemed like a nice defensive step against an unnecessary query.

I'm going to try extracting a helper method out of this code block. Then a return-early pattern will probably clear things up.